### ISSUE TITLE
bugs for First Person Body Support (#1396)

### ISSUE THREAD

**Issue by yohjimane** [Swartz27/xray-16#1421]
All issues below effect branch #1396 and must be completed before we can merge. 

- [x] 1. mixed build errors, can cause first person body to appear floating. The second error occurs on dev branch in mixed builds too.
- No longer an issue with new camera positioning method 9dd1ae265d8ff4876a8cc3dfe2c8b1b989d6154a

![image](https://github.com/OpenXRay/xray-16/assets/11345783/08d0e476-2b7d-452b-9445-873a35c40317)
![image](https://github.com/OpenXRay/xray-16/assets/11345783/29dc6610-a82e-4ece-b8dc-fbd302f60d2a)

___

- [x] 2. being around NPCs causes first person body position to deviate
- fixed with 900c3270eba067e7ac24c08ec1a02ee53f90286e

![xrEngine_2023_07_11_17_44_56_641](https://github.com/OpenXRay/xray-16/assets/11345783/030dbb67-f04d-4e2d-ab98-cb76a60362b8)

___

- [x] 3. possible to clip through object if facing away from them
- No longer an issue new camera positioning method 9dd1ae265d8ff4876a8cc3dfe2c8b1b989d6154a

![xrEngine_2023_07_11_17_37_23_774](https://github.com/OpenXRay/xray-16/assets/11345783/243e3ecc-aaf6-4391-9ee5-a87acc63df95)

___

- [x] 4. First person body position can "jitter" between original position and corrected position
- fixed with 60ac2ea0fcbe1f6ce5700ba705ba015d70b78abd
- More info in [discord msg](https://discord.com/channels/410170555619082240/489453264157409281/1137364447455875072)

https://github.com/OpenXRay/xray-16/assets/11345783/ee1ac41c-4486-44b0-bace-916f6a16a68e

___

- [x] 5. New jitter bug introduced with 60ac2ea0fcbe1f6ce5700ba705ba015d70b78abd
- fixed with fcbac9fb0adbc78643f25b963e7cf5c8199fabb4

https://github.com/OpenXRay/xray-16/assets/11345783/8b4774a1-23a8-4ec2-9a6c-23ab96a110d8

___

- [x] 6. First person body syncs at low frame rate in some conditions. I think it depends on whether or not our body can currently cast a shadow. Possibly some performance optimization we need to disable. 
- fixed with c32e8662b700ca04089d774513ffbc97abcb1c7e

- Can be easily reproduced with the settings: `r__actor_shadow off` and `g_first_person_body on`
- In the video below, I am turning headlamp on, which causes the body to render smoothly. When it is off, the body renders at lower frame rate

https://github.com/OpenXRay/xray-16/assets/11345783/e9703f25-9b5d-4c62-baa3-11dc44fad134

___

- [x] 7. Actor headlamp is in incorrect position
- It needs to be corrected similarly to [CWeapon position correction](https://github.com/OpenXRay/xray-16/pull/1396/files#diff-6219d0026933a729c3dd9c14ae7548213be8bd9117c2c9291b61db2d48e769e7R191)
- Would be best if we can find a generic solution that works for CWeapon, CTorch, CCustomDetector etc...
- fixed with new camera positioning method 9dd1ae265d8ff4876a8cc3dfe2c8b1b989d6154a

![image](https://github.com/OpenXRay/xray-16/assets/11345783/052c1e18-8063-4857-90e0-21cc1dc3fa48)

___
- [x] 8. IK is incorrectly based on original position
- It should be corrected using camera direction similarly to CWeapon position correction.
- fixed with new camera positioning method 9dd1ae265d8ff4876a8cc3dfe2c8b1b989d6154a

![xrEngine_2023_08_19_02_40_31_861](https://github.com/OpenXRay/xray-16/assets/11345783/e360fa6e-d6ac-41da-9b6a-c9ff0a63b7cc)

**Comment by yohjimane** [comment]
The last of the bugs have been resolved due to the newer camera implementation 5749b5db2f7b873c889d4b30069e89cd695f4471

Closing this out. There is one remaining bug, but it could be considered both first person and third person bug, so i will make a new ticket for it :)
